packages/nimble-components/src/select/index.ts

```diff
@@ -869,21 +897,27 @@ export class Select
         };
         let selectedIndex = -1;
         let firstValidOptionIndex = -1;
+        let placeholderIndex = -1;
         for (let i = 0; i < options?.length; i++) {
-            const option = options[i];
-            if (optionIsSelected(option!) || option?.value === this.value) {
+            const option = options[i]!;
+            if (optionIsSelected(option) || option.value === this.value) {
                 selectedIndex = i;
-            }
-            if (
+                break;
+            } else if (placeholderIndex === -1 && isOptionPlaceholder(option)) {
+                placeholderIndex = i;
+            } else if (
                 firstValidOptionIndex === -1
-                && isOptionSelectable(option! as ListOption)
+                && placeholderIndex === -1
+                && isOptionSelectable(option as ListOption)
             ) {
                 firstValidOptionIndex = i;
             }
         }
 
         if (selectedIndex !== -1) {
             this.selectedIndex = selectedIndex;
+        } else if (placeholderIndex !== -1) {
+            this.selectedIndex = placeholderIndex;
         } else if (firstValidOptionIndex !== -1) {
             this.selectedIndex = firstValidOptionIndex;
         } else {
```

Comment:
No, despite the `Select` now being clearable, a value of -1 is not considered a valid default state. This is how the native `select` is spec'd.

Reply:
[MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/selectedIndex) says -1 is valid:
![image](https://github.com/ni/nimble/assets/7282195/ee8641ce-fdb3-4a11-af5a-e99146f66df5)

I think it would have to be. If there's no placeholder and the select is cleared, then any other value for `selectedIndex` would be a lie, right?

Reply:
I'm actually going to backtrack on what I said...the HTML `select` spec says the following:
"If element's [multiple](https://html.spec.whatwg.org/multipage/form-elements.html#attr-select-multiple) attribute is absent, and element's [display size](https://html.spec.whatwg.org/multipage/form-elements.html#concept-select-size) is 1, and no [option](https://html.spec.whatwg.org/multipage/form-elements.html#the-option-element) elements in the element's [list of options](https://html.spec.whatwg.org/multipage/form-elements.html#concept-select-option-list) have their [selectedness](https://html.spec.whatwg.org/multipage/form-elements.html#concept-option-selectedness) set to true, then set the [selectedness](https://html.spec.whatwg.org/multipage/form-elements.html#concept-option-selectedness) of the first [option](https://html.spec.whatwg.org/multipage/form-elements.html#the-option-element) element in the [list of options](https://html.spec.whatwg.org/multipage/form-elements.html#concept-select-option-list) in [tree order](https://dom.spec.whatwg.org/#concept-tree-order) that is not [disabled](https://html.spec.whatwg.org/multipage/form-elements.html#concept-option-disabled), if any, to true, and return."

I take this to mean that if _all_ options are disabled (and none have `hidden` and `selected` set as well), then it can, indeed be set to -1. I'm going to make this change.